File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.java
class Solution {
    public int findMin(int[] nums) {
        int min=nums[0];
        int start=0,end=nums.length-1;
        while(start<=end){
            if(nums[start]<min){
                min=nums[start];
                break;
            }
            int mid=start+(end-start)/2;
            if(nums[mid]>=nums[start]){
                start=mid+1;
            }
            else{
                end=mid;
            }
        }
        return min;
    }
}

File: Data Structures & Algorithms/majority-element/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 0
        el = 0

        for n in nums:
            if count == 0:
                count = 1
                el = n
            elif el == n:
                count += 1
            else:
                count -= 1
        
        return el